builder/shell-for.nix

```diff
@@ -9,6 +9,7 @@
 , additional ? _: []
 , withHoogle ? true
 , exactDeps ? false
+, tools ? t: []
```

Comment:
I did this mostly to match the ghc version used to build the tool with the one in the shell.  Some tools (like `ghcide` need it to match), and for others I think it might reduce the foot print.

Reply:
I think for most things it doesn't matter but yeah, the IDE tools are fussy about that. It just feels a little unpleasant to have it be special here...